hooks/useSwitchNetwork.ts

```diff
@@ -0,0 +1,35 @@
+import { useNetwork, useSwitchNetwork as useWagmiSwitchNetwork } from 'wagmi'
+
+import { gaEvent } from 'lib/gtag'
+import { convertChainIdToHex, networkChainId } from 'utils/network'
+
+export function useSwitchNetwork() {
+  const { chain } = useNetwork()
+  const { switchNetwork: initSwitchNetwork } = useWagmiSwitchNetwork({
+    chainId: networkChainId,
+    throwForSwitchChainNotSupported: true,
+  })
+
+  function switchNetwork() {
+    try {
+      initSwitchNetwork?.(networkChainId)
+    } catch {
+      if (!window.ethereum) return
+      window.ethereum.request({
```

Comment:
Can I get the reasoning behind manual interaction with window.ethereum? Because as far as I know this won't work with WalletConnect.

Reply:
I intentionally added manual interaction with `window.ethereum` to handle an edge case: `switchNetwork` via `wagmi` fails when user is not connected.
But it is true that it cannot deal with users with WalletConnect, so I should find another way to address this.

_▼ This is an error message when user tries to switch network without connecting his/her wallet:_

<img width="438" alt="image" src="https://user-images.githubusercontent.com/19285811/193005440-a303ecc3-5f96-47d6-b469-42113d67b106.png">

Reply:
Cool workaround.